java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java

```diff
@@ -465,6 +466,100 @@ public boolean equals(@NullAllowed final Object other) {
                     new DynLkp());
         }
 
+        @Override
+        public boolean canDestroy() {
+            return true;
+        }
+
+        @Override
+        public void destroy() {
```

Comment:
Why did the delete routine grow from 20 lines (original commit to nearly 100 lines)?

Reply:
@matthiasblaesing, I haven't deregister classes, packages like it does in package deletion in previous commit, I have done the same in current commit.

Reply:
But is this all necessary? I can delete files outside the IDE and my experience with netbeans is, that it correctly detects this and updates the state of the projects.